### ISSUE TITLE
[clang-tidy] `portability-template-virtual-member-function` warns about pure virtual member functions

### ISSUE THREAD

**Issue by tearfur** [llvm/llvm-project#139031]
`portability-template-virtual-member-function` warns about pure virtual functions that are overidden and implemented, and the overridden method is used.

Is this a false alarm? If not, how am I supposed to fix this warning?

https://godbolt.org/z/zaT3TWbbb

```
<source>:10:18: warning: unspecified virtual member function instantiation; the virtual member function is not instantiated but it might be with a different compiler [portability-template-virtual-member-function]
   10 |     virtual void foo() = 0;
      |                  ^
<source>:13:25: note: template instantiated here
   13 | struct Derived : public Base<int>
      |
```

Current trunk: fc8484f0e383cc5cf31d67ad3e762705955ea1ea

**Comment by tearfur** [comment]
CC @isuckatcs as the author of `portability-template-virtual-member-function` #110099.
